File: src/main/resources/static/lib/js/classgrades.js
//注意：导航 依赖 element 模块，否则无法进行功能性操作
layui.use('element', function(){
    var element = layui.element;

    //…
});

$(function () {

    //查看学生信息

    studentdetail = function(data){
        layui.use('layer', function(){
            var layer = layui.layer;

            layer.open({

                title: ['学生信息'],
                type: 2,
                content: 'page/student/studentdetail.html',
                shadeClose: true,
                area: ['1000px', '700px'],
                success: function (layero,index) {
                    console.log(data);
                    var body = layui.layer.getChildFrame('body', index);
                    body.find(".stuId").text(data.stuId);
                    body.find(".stuName").text(data.stuName);
                    body.find(".stuPassword").text(data.stuPassword);
                    body.find(".stuEmail").text(data.stuEmail);
                    body.find(".stuIdCard").text(data.stuIdCard);
                    body.find(".stuSex").text(data.stuSex);
                    body.find(".stuBirthday").text(data.stuBirthday);
                    body.find(".nationName").text(data.nationName);
                    body.find(".stuStatus").text(data.stuStatus);
                    body.find(".stuAge").text(data.stuAge);
                    body.find(".classId").text(data.classId);
                    body.find(".stuenRollmentTime").text(data.stuenRollmentTime);
                    body.find(".stuPolitical").text(data.stuPolitical);
                    body.find(".stuAddress").text(data.stuAddress);
                    body.find(".stuImage").text(data.stuImage);



                    layui.form.render();
                },
                cancel:function () {

                },
                end:function () {



                }
            });

        });
    }



});
layui.use(['laydate', 'laypage', 'layer', 'table', 'carousel', 'upload', 'element', 'slider'], function(){
    var laydate = layui.laydate //日期
        ,laypage = layui.laypage //分页
        ,layer = layui.layer //弹层
        ,table = layui.table //表格
        ,carousel = layui.carousel //轮播
        ,upload = layui.upload //上传
        ,element = layui.element //元素操作
        ,slider = layui.slider //滑块



    var stuId = $("#stuId").text();
    console.log("id："+stuId);

    //执行一个 table 实例
    table.render({
        elem: '#demo'
        ,height: 500
        ,url: 'classgrades' //数据接口
        ,title: '用户表'
        ,page: true //开启分页
        ,title:'学生信息表'
        ,toolbar: 'true' //开启工具栏，此处显示默认图标，可以自定义模板，详见文档

        ,cols: [ [ //表头
            {type: 'checkbox', fixed: 'left'}
            // ,{field: 'stuId', title: '学号', width:'10%', sort: true}
            // ,{field: 'stuName', title: '姓名', width:'8%'}
            ,{field: 'classId', title: '班级', width:'8%', sort: true}
            ,{field: 'testTime', title: '考试时间', width:'8%', sort: true}
            ,{field: 'testDescribe', title: '描述', width:'8%', sort: true}


            ,{field: 'totalMax', title: '总分最高分', width:'9%', sort: true}
            ,{field: 'totalMin', title: '总分最低分', width:'9%', sort: true}
            ,{field: 'totalAve', title: '总分平均分', width:'9%', sort: true}


            ,{field: 'chineseMax', title: '语文最高分', width:'9%', sort: true}
            ,{field: 'chineseMin', title: '语文最低分', width:'9%', sort: true}
            ,{field: 'chineseAve', title: '语文平均分', width:'9%', sort: true}

            ,{field: 'mathematicsMin', title: '数学最高分', width:'9%', sort: true}
            ,{field: 'mathematicsMax', title: '数学最低分', width:'9%', sort: true}
            ,{field: 'mathematicsAve', title: '数学平均分', width:'9%', sort: true}

            ,{field: 'englishMax', title: '英语最高分', width:'9%', sort: true}
            ,{field: 'englishMin', title: '英语最低分', width:'9%', sort: true}
            ,{field: 'englishAve', title: '英语平均分', width:'9%', sort: true}

            ,{field: 'politicsMax', title: '政治最高分', width:'9%', sort: true}
            ,{field: 'politicsMin', title: '政治最低分', width:'9%', sort: true}
            ,{field: 'politicsAve', title: '政治平均分', width:'9%', sort: true}

            ,{field: 'historyMax', title: '历史最高分', width:'9%', sort: true}
            ,{field: 'historyMin', title: '历史最低分', width:'9%', sort: true}
            ,{field: 'historyAve', title: '历史平均分', width:'9%', sort: true}

            ,{field: 'geographyMax', title: '地理最高分', width:'9%', sort: true}
            ,{field: 'geographyMin', title: '地理最低分', width:'9%', sort: true}
            ,{field: 'geographyAve', title: '地理平均分', width:'9%', sort: true}


            ,{field: 'biologyMax', title: '生物最高分', width:'9%', sort: true}
            ,{field: 'biologyMin', title: '生物最低分', width:'9%', sort: true}
            ,{field: 'biologyAve', title: '生物平均分', width:'9%', sort: true}

            ,{field: 'chemistryMax', title: '化学最高分', width:'9%', sort: true}
            ,{field: 'chemistryMin', title: '化学最低分', width:'9%', sort: true}
            ,{field: 'chemistryAve', title: '化学平均分', width:'9%', sort: true}

            ,{field: 'physicsMax', title: '物理最高分', width:'9%', sort: true}
            ,{field: 'physicsMin', title: '物理最低分', width:'9%', sort: true}
            ,{field: 'physicsAve', title: '物理平均分', width:'9%', sort: true}

            ,{field: 'musicMax', title: '音乐最高分', width:'9%', sort: true}
            ,{field: 'musicMin', title: '音乐最低分', width:'9%', sort: true}
            ,{field: 'musicAve', title: '音乐平均分', width:'9%', sort: true}

            ,{field: 'artsMax', title: '美术最高分', width:'9%', sort: true}
            ,{field: 'artsMin', title: '美术最低分', width:'9%', sort: true}
            ,{field: 'artsAve', title: '美术平均分', width:'9%', sort: true}



            ,{field: 'sportsMax', title: '体育最高分', width:'9%', sort: true}
            ,{field: 'sportsMin', title: '体育最低分', width:'9%', sort: true}
            ,{field: 'sportsAve', title: '体育平均分', width:'9%', sort: true}

            // ,{fixed: 'right', width: '9%', align:'center', toolbar: '#barDemo'}

        ] ]
    });


    //监听行工具事件
    table.on('tool(test)', function(obj){ //注：tool 是工具条事件名，test 是 table 原始容器的属性 lay-filter="对应的值"
        var data = obj.data //获得当前行数据
            ,layEvent = obj.event; //获得 lay-event 对应的值
        if(layEvent === 'detail'){
            var classId = obj.data.classId;
            var url = 'class/detail/'+classId;
            console.log(url);

            $(window).attr("location",url);





        }
    });

//
});

File: src/main/java/com/braisedpanda/shirotest/model/po/ClassGrades.java
package com.braisedpanda.shirotest.model.po;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Id;
import javax.persistence.Table;
import java.io.Serializable;

@Data
@Table(name="classgrades")
public class ClassGrades implements Serializable{

    private static final long serialVersionUID = 1536811172206548570L;
    @Id
    @Column(name="classGradesId")
    private String classGradesId;      //classGradesId
    @Column(name="classGradesCardId")
    private String classGradesCardId; //根据classGradesCardId，作为查找入口
    @Column(name="totalAve")
    private Double totalAve;             //各项班级总分、平均分、最低分、最高分
    @Column(name="totalMax")
    private Double totalMax;
    @Column(name="totalMin")
    private Double totalMin;
    @Column(name="chineseMax")
    private Double chineseMax;
    @Column(name="chineseMin")
    private Double chineseMin;
    @Column(name="chineseAve")
    private Double chineseAve;
    @Column(name="mathematicsMin")
    private Double mathematicsMin;
    @Column(name="mathematicsMax")
    private Double mathematicsMax;
    @Column(name="mathematicsAve")
    private Double mathematicsAve;
    @Column(name="englishMax")
    private Double englishMax;
    @Column(name="englishMin")
    private Double englishMin;
    @Column(name="englishAve")
    private Double englishAve;
    @Column(name="politicsMin")
    private Double politicsMin;
    @Column(name="politicsMax")
    private Double politicsMax;
    @Column(name="politicsAve")
    private Double politicsAve;
    @Column(name="historyMax")
    private Double historyMax;
    @Column(name="historyMin")
    private Double historyMin;
    @Column(name="historyAve")
    private Double historyAve;
    @Column(name="geographyMin")
    private Double geographyMin;
    @Column(name="geographyAve")
    private Double geographyAve;
    @Column(name="geographyMax")
    private Double geographyMax;
    @Column(name="biologyMax")
    private Double biologyMax;
    @Column(name="biologyMin")
    private Double biologyMin;
    @Column(name="biologyAve")
    private Double biologyAve;
    @Column(name="chemistryMin")
    private Double chemistryMin;
    @Column(name="chemistryAve")
    private Double chemistryAve;
    @Column(name="chemistryMax")
    private Double chemistryMax;
    @Column(name="physicsMin")
    private Double physicsMin;
    @Column(name="physicsAve")
    private Double physicsAve;
    @Column(name="physicsMax")
    private Double physicsMax;
    @Column(name="musicMin")
    private Double musicMin;
    @Column(name="musicAve")
    private Double musicAve;
    @Column(name="musicMax")
    private Double musicMax;
    @Column(name="artsMin")
    private Double artsMin;
    @Column(name="artsAve")
    private Double artsAve;
    @Column(name="artsMax")
    private Double artsMax;
    @Column(name="sportsMin")
    private Double sportsMin;
    @Column(name="sportsMax")
    private Double sportsMax;
    @Column(name="sportsAve")
    private Double sportsAve;


}
